dashboard/www/scripts/utils/utils.js

```diff
@@ -4,26 +4,51 @@ window.discotron.utils = class {
      * @param {string} verb HTTP verb to send request with
      * @param {string} url Url to make the post request on
      * @param {object} [body] Data that will be JSON.stringified and sent to the website. Does not work for GET requests.
+     * @param {object} [appToken] Application token that will be sent in the Authorization header
      * @returns {Promise} resolve(data {object|string}) data: object if could parse JSON, reject()
      */
-    static async query(verb, url, body) {
-        if (verb === "GET" && body !== undefined) {
-            throw new Error("GET cannot have a body.");
+    static async query(verb, url, body, appToken) {
+        const headers = { "Content-Type": "application/json" };
+
+        if (appToken !== undefined) {
+            headers["Authorization"] = `Bearer ${appToken}`;
         }
 
-        const response = await fetch(url, {
+        const request = {
             method: verb,
+            headers: headers,
             mode: "cors",
             cache: "no-cache",
             credentials: "same-origin",
-            headers: {
-                "Content-Type": "application/json"
-            },
             redirect: "follow",
             referrerPolicy: "no-referrer",
-            body: JSON.stringify(body)
-        });
-        return await response.json();
+        };
+
+        // Get cannot have a body, but we want to easily be able to use an object in the parameters
+        if (verb === "GET") {
+            const parameters = {};
+            for (const key in body) {
+                const element = body[key];
+
+                // We sadly convert everything to JSON, as we may want to be able to use more complex objects
+                parameters[key] = JSON.stringify(element);
+            }
+
+            url += "?" + new URLSearchParams(parameters).toString();
+            body = undefined; // Unset the body as get cannot have one
+        }
+
+        if (body !== undefined) {
+            request.body = JSON.stringify(body);
+        }
+
+        const response = await fetch(url, request);
+        try {
+            return await response.json();
+        } catch (error) {
+            console.error("API Query failed for URL", response);
+            throw new discotron.WebApiError("An error occurred when communicating with the bot. Status code: " + response.status);
```

Comment:
I'm kind of against using a WebApiError in a general `utils` function. Even though it's ugly and annoying, we should possibly re-catch on the calling side, and use a regular `Error` type here?